File: LeetCode/Simulation/Python/Leetcode1784-check-if-binary-string-has-at-most-one-segment-of-ones.py
class Solution:
    def checkOnesSegment(self, s: str) -> bool:
        cnt,flag = 0,False
        for i in s:
            if i == '1':
                if flag: continue
                else:
                    cnt += 1
                    if cnt > 1: return False
                    flag = True
            else:
                flag = False
        return True

File: LeetCode/Simulation/Rust/Leetcode1784-check-if-binary-string-has-at-most-one-segment-of-ones.rs
impl Solution {
  pub fn check_ones_segment(s: String) -> bool {
      let s = s.as_bytes();
      let (mut cnt,mut flag) = (0,false);
      for &c in s.iter() {
          if c == b'1' {
              if flag {
                  continue;
              } else {
                  cnt += 1;
                  if cnt > 1 {
                      return false;
                  }
                  flag = true;
              }
          } else {
              flag = false;
          }
      }
      true
  }
}